validator/distro_validator_test.go

```diff
@@ -360,16 +364,25 @@ func TestEnsureValidContainerPool(t *testing.T) {
 	assert.Nil(err)
 }
 
-func TestEnsureValidBootstrapMethod(t *testing.T) {
+func TestEnsureValidBootstrapAndCommunicationMethods(t *testing.T) {
 	ctx := context.Background()
-	for _, bootstrapMethod := range []string{
-		distro.BootstrapMethodLegacySSH,
-		distro.BootstrapMethodSSH,
-		distro.BootstrapMethodPreconfiguredImage,
-		distro.BootstrapMethodUserData,
-	} {
-		assert.Nil(t, ensureValidBootstrapMethod(ctx, &distro.Distro{BootstrapMethod: bootstrapMethod}, &evergreen.Settings{}))
+	for _, bootstrapMethod := range distro.ValidBootstrapMethods {
+		for _, communicationMethod := range distro.ValidCommunicationMethods {
+			d := &distro.Distro{BootstrapMethod: bootstrapMethod, CommunicationMethod: communicationMethod}
+			if bootstrapMethod == distro.BootstrapMethodLegacySSH && communicationMethod == distro.CommunicationMethodLegacySSH {
```

Comment:
switch statement? this logic. seems a bit hard to follow.

Reply:
Done.